tensordict/prototype/tensorclass.py

```diff
@@ -340,8 +354,16 @@ def _setitem(self, item, value):
         raise ValueError("Invalid indexing arguments.")
     if not isinstance(value, self.__class__):
         raise ValueError("__setitem__ is only allowed for same-class assignement")
+    # Validating the non-tensor data before setting the item
     for key, val in self.non_tensordict.items():
-        if not isinstance(val, _accepted_classes):
+        # Need the nested tensor class to be the same
+        if is_tensorclass(val):
+            if (val != value.non_tensordict[key]).all():
+                raise ValueError(
+                    f"Expecting {repr(val)} for {key} instead got {repr(value.non_tensordict[key])}"
```

Comment:
give a bit more of context of what we are doing. repr could be huge and not so meaningful